projects/plugins/jetpack/_inc/client/recommendations/summary/index.jsx

```diff
@@ -39,16 +40,20 @@ const SummaryComponent = props => {
 		isFetchingSidebarData,
 		sidebarCardSlug,
 		siteTitle,
+		siteRawUrl,
+		siteAdminUrl,
 		summaryFeatureSlugs,
 		updateRecommendationsStep,
-		upgradeUrl,
 		upsell,
 	} = props;
 
 	useEffect( () => {
 		updateRecommendationsStep( 'summary' );
 	}, [ updateRecommendationsStep ] );
 
+	const upgradeUrl = upsell.product_slug
+		? generateCheckoutLink( upsell.product_slug, siteAdminUrl, siteRawUrl )
+		: null;
```

Comment:
In theory, since `upsell` is loaded into the state asynchronously, there's a chance that it may just be an empty object when the component renders - if the request were to be slow/ fail.

If empty, that could yield `null` here (and also have an effect on the `<ProductCardUpsellComponent>`). However, I think checking to make sure that `upsell` is defined/ loaded before rendering the related sidebar card is not within the scope of this PR - maybe we can follow up on that in a separate issue?